frontend/src/rauta.tsx

```diff
@@ -269,19 +303,45 @@ export const makeRouter = <C extends Config, >(config: C): RouterLib => {
             // Whenever the user navigates forwards or backwards in the browser,
             // we have to render the corresponding route. We also restore the
             // scroll position which we store within the history state.
+            // Finally, this is used to prevent navigating away from a route if
+            // this is blocked.
+            let ignoreNextPop = false;
             const onPopState = (e: PopStateEvent) => {
+                if (ignoreNextPop) {
```

Comment:
I have no idea. I thought about this as well. But I can't think of any other way to prevent this "own triggering", the `history` package does it basically the same, and it worked for me. At least the popup prevents me from pressing back/forward again.

Reply:
Couldn't we put something in the state to recognize the event caused by our own code? :thinking:

Reply:
Not really. We can only modify the current state. So when we say `go` we cannot modify the target state. Similarly, in the next `onpopstate` call, we cannot check the state we came from.

Reply:
For the record: There was further discussion, and it's a hard problem. We will leave it as is for now, and there is a comment now. :+1: